1.Console-EchoBot for NODEJS

benbrown opened the pull request:
A simple console-based echobot example.

Even with this simple example, I thought we should start with the folder and file structure in place. Is it strictly necessary to pull the echo feature into the mainDialog? No, but its the right model.

Additionally, I put in the .env loader, even though it isn't yet used.

- [ ] I need help with the deploymentScripts.
- [ ] There is a link to docs for TurnContext objects that I copied from an old sample that seems broken.

vishwacsena commented on javascript_nodejs/1.console-echobot/package.json:
```diff
@@ -0,0 +1,20 @@
+  "scripts": {
+    "start": "node ./index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
For consistency, we should add "watch": "nodemon ./index.js"

benbrown replied:
OK!

vishwacsena commented on javascript_nodejs/1.console-echobot/index.js:
```diff
@@ -0,0 +1,27 @@
+const mainDialog = require('./dialogs/mainDialog');
+
+// load environment variables from .env file.
+const ENV_FILE = path.join(__dirname, '..', '.env');
```
Your .env file would typically be at root so unclear if we need '..' here. Would be better with path.join(__dirname, '.env');

cleemullins commented on javascript_nodejs/1.console-echobot/dialogs/mainDialog/index.js:
```diff
@@ -0,0 +1,19 @@
+    // (it could theoretically be another type of activity)
+    if (context.activity.type === 'message') {
+
+        // check to see if the user sent a simple "quit" message
```
Extra blank line

cleemullins commented on javascript_nodejs/1.console-echobot/README.md:
```diff
@@ -0,0 +1,51 @@
+
+Adapters such as the `ConsoleAdapter` provide an abstraction for your bot to work with a variety of environments and messaging platforms.
+
+The adapter is responsible for directing incoming and outgoing communication, authentication, and so on. Adapters for different platforms and
```
"and so on" isn't a good way to end a sentence.

cleemullins commented on javascript_nodejs/1.console-echobot/README.md:
```diff
@@ -0,0 +1,51 @@
+When your bot receives an activity, the adapter wraps up everything about that activity, creates a [context object](https://github.com/Microsoft/botbuilder-js/blob/master/doc/botbuilder/classes/botbuilder.turncontext.md), passes it to your bot's application logic,
+and sends responses generated by your bot back to the user's channel.
+
+In most situations, we don't work with the adapter directly, but it's good to know it's there and what it does. In this example, we're using the
```
who is "we" in this case? Consider removing or changing the articles.

cleemullins replied:
Strongly consider removing the entire first sentence in this paragraph.

cleemullins commented on javascript_nodejs/1.console-echobot/README.md:
```diff
@@ -0,0 +1,51 @@
+and sends responses generated by your bot back to the user's channel.
+
+In most situations, we don't work with the adapter directly, but it's good to know it's there and what it does. In this example, we're using the
+`ConsoleAdapter`, which provides a very simple way to exchange messages - they're sent and received via the terminal window. Using the ConsoleAdapter
```
"very simple" or "simple". No need for the "very". simplify.

cleemullins replied:
"console window"

cleemullins commented on javascript_nodejs/1.console-echobot/README.md:
```diff
@@ -0,0 +1,51 @@
+
+In most situations, we don't work with the adapter directly, but it's good to know it's there and what it does. In this example, we're using the
+`ConsoleAdapter`, which provides a very simple way to exchange messages - they're sent and received via the terminal window. Using the ConsoleAdapter
+is a great way of getting started quickly with Botbuilder. It also allows you to build and test bots that operate on your local machine without any
```
BotBuilder vs Botbuilder.

cleemullins replied:
"It also" isn't a good way to start a sentence. Consider simplifying the grammar.

cleemullins commented on javascript_nodejs/1.console-echobot/README.md:
```diff
@@ -0,0 +1,51 @@
+is a great way of getting started quickly with Botbuilder. It also allows you to build and test bots that operate on your local machine without any
+external API calls or client applications.
+
+Other available adapters can connect your bot to the web, or through the BotFramework Connector Service, to many popular messaging platforms.
```
remove "can". Simplify language.

cleemullins commented on javascript_nodejs/1.console-echobot/index.js:
```diff
@@ -0,0 +1,27 @@
+const ENV_FILE = path.join(__dirname, '..', '.env');
+const loadFromEnv = config({path: ENV_FILE});
+
+// create the Bot adapter, which is responsible for sending and receiving messages
```
"Create" vs "create"

cleemullins replied:
add a "." at the end.

cleemullins commented on javascript_nodejs/1.console-echobot/index.js:
```diff
@@ -0,0 +1,27 @@
+const loadFromEnv = config({path: ENV_FILE});
+
+// create the Bot adapter, which is responsible for sending and receiving messages
+// in this case, we are using the ConsoleAdapter, which enables a bot you can chat with from within your terminal window.
```
remove "in this case". Simplify language.

cleemullins commented on javascript_nodejs/1.console-echobot/index.js:
```diff
@@ -0,0 +1,27 @@
+// in this case, we are using the ConsoleAdapter, which enables a bot you can chat with from within your terminal window.
+const adapter = new botbuilder.ConsoleAdapter();
+
+// tell the adapter to start listening for incoming messages and events, known as "activities"
```
Make these sentences.

cleemullins commented on javascript_nodejs/1.console-echobot/package.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "console-echobot",
+  "version": "1.0.0",
+  "description": "Bot Builder v4 console echobot sample",
```
BotBuilder vs Bot Builder

benbrown commented on javascript_nodejs/1.console-echobot/README.md:
````diff
@@ -0,0 +1,51 @@
+    ```bash
+    cd javascript_nodejs/1.console-echobot
+    ```
+    - Point to the myget feed 
````
will fix. FYI, this is copy-pasted from the sample typescript project, so we should probably fix it there too before other people copy into their own projects.

benbrown commented on javascript_nodejs/1.console-echobot/README.md:
````diff
@@ -0,0 +1,51 @@
+# To try this sample
+- Clone the repository
+    ```bash
+    git clone https://github.com/Microsoft/botbuilder-samples.git
````
noted.

benbrown commented:
@vishwacsena @cleemullins I believe I have addressed all of your feedback. Thanks!

As I mentioned, some of the little grammar style issues @cleemullins pointed out were copy-pasted from [here](https://github.com/Microsoft/BotBuilder-Samples/tree/v4/javascript_ts/2.echobot-with-counter), so we should update there as well
